mdp/nodes/misc_nodes.py

```diff
@@ -815,3 +826,603 @@ def _execute(self, x):
+
+    The list of available transformation function names can be accessed by calling
+    'get_transform_fn_names' method.
+
+    To add other transformation methods, subclasses can overwrite the method
+    '_get_transform_fns' that returns a dict mapping labels to the transformation methods.
+
+    A few of the transformations might need additional arguments. Set 'transform_seq_args' variable equal
+    to the list of argument-tuples in the same order as the 'transform_seq'. Use None whenever an
+    intermediate transformation function requires no arguments. In any case, the 'transform_seq_args'
+    setter raises an exception if the required number of arguments are not passed.
+
+    An example implementation of the node:
+
+    node = TransformerNode(input_shape=(640, 480, 3), transform_seq=['set_shape', 'gray', 'img_255_1', 'center'])
+
+    Let the input to the node be a block of 10 flattened RGB images (shape - (10, 640*480*3)).
+    The input is first reshaped to the given inp_shape since no argument has been passed (shape - (10, 640, 480, 3)).
+    It is then converted to grayscale (shape - (10, 640, 480)).
+    The output is then rescaled from [0, 255] to [0, 1], and then the data is centered to have zero mean.
+    The final output returned is a flattened 2d array (shape - (10, 640*480*3)).
+
+    The node also includes sklearn's pre-processing functions if sklearn package is available.
+
+    """
+
+    def __init__(self, input_shape, transform_seq=None, transform_seq_args=None, input_dim=None, dtype=None):
+        """
+        input_shape - The shape of each input sample. E.g. for a video of ten 640 x 480 images, the input_shape
+                        is (640, 480). The input data is always first transformed to this shape.
+        transform_seq - A list of strings, where each string represents a label for the transformation.
+                        Supported transformation:
+                            'transpose' - transposes axis dimensions > 0
+                            'center' - centers the data (removes mean over axis=0)
+                            'resize' - resize data dims (for image data). Requires python-opencv or pil packages.
+                            'img_255_1' - scales uint img data to float values between [0, 1]
+                            'gray' - converts to grayscale images.
+                            'to_2d' - converts data to 2D
+                            'to_dtype' - converts data to the desired data type
+                            'set_shape' - reshapes the data to the given shape. If shape argument is not provided,
+                            then it uses a stored shape buffer, which is either the input_shape or the shape
+                            lost if 'to_2d' is executed.
+
+        trasnform_seq_args - A list of required arguments tuples for each transformation, ordered according to
+                            the transform_seq.
```

Comment:
This sound to me like a candidate for the strategy pattern, or different subclasses (similar to the situation with the BasisFunctionNode).